feat: import obj-c-extensions as class members (#3476)
JetBrains/kotlin-web-site: docs/topics/native/native-objc-interop.md

The names of Kotlin classes and interfaces are prefixed when imported to Objective-C.
The prefix is derived from the framework name.

> Objective-C does not support packages in a framework. Thus, the Kotlin compiler renames Kotlin classes, which have the
> same name but different package, in the same framework. This algorithm is not stable yet and can change between Kotlin
> releases. As a workaround, you can rename the conflicting Kotlin classes in the framework.

Reviewer: ```suggestion
Objective-C does not support packages in a framework. If the Kotlin compiler finds Kotlin classes in the same framework which have the same name but different packages, it renames them. To work around this, you can rename the conflicting Kotlin classes in the framework.
```

which throws an exception to be handled, then the Kotlin method should be marked
with a `@Throws` annotation specifying a list of "expected" exception classes.

> When compiling to the Objective-C/Swift framework, non-`suspend` functions having or inheriting

Reviewer: ```suggestion
When compiling to the Objective-C/Swift framework, non-`suspend` functions that have or inherit the
```

### Extensions and category members

Members of Objective-C categories and Swift extensions are generally imported to Kotlin
as extensions. That's why these declarations can't be overridden in Kotlin.
And the extension initializers aren't available as Kotlin constructors.

> > Currently, there are two exceptions in UIKit. Starting with Kotlin 1.8.20, category members that are declared

Reviewer 2: NSView is from AppKit and UIView is from UIKit.

### Extensions and category members

Members of Objective-C categories and Swift extensions are generally imported to Kotlin
as extensions. That's why these declarations can't be overridden in Kotlin.
And the extension initializers aren't available as Kotlin constructors.

> Currently, there are two exceptions. Starting with Kotlin 1.8.20, category members that are declared

> > in the same headers as the NSView class (from the AppKit framework) and UIView classes (from the UIKit framework) are

Reviewer: Since you say there are two exceptions, should it be: "headers as the NSView class ... **or** UIView classes"?

Author: I agree, let't use 'or' in all cases

> > imported as members of these classes. It means that you can override methods subclassing from NSView and UIView.

Reviewer: ```suggestion
> imported as members of these classes. This means that you can override methods when subclassing from NSView and UIView.
```

Author: The initial idea was to say "you can override methods that subclass from NSView and UIView." but maybe your variant works too

Known limitations:
* Dependency documentation is not exported unless it is compiled with `-Xexport-kdoc` itself. The feature is experimental, 
so libraries compiled with this flag might be incompatible with other compiler versions.

> * KDoc comments are mostly exported "as is", many KDoc features (for example, `@property`) are not supported.

Reviewer: ```suggestion
* KDoc comments are mostly exported "as is". Many KDoc features (for example, `@property`) are not supported.
```
